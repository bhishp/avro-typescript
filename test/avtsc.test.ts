import * as fs from "fs";
import { avroToTypeScript, RecordType } from "../src";
import { Schema } from "../src/model";

describe("avroToTypeScript", () => {
    test("it should generate an interface", () => {
        const schemaText = fs.readFileSync(__dirname + "/example.avsc", "utf-8");
        const schema = JSON.parse(schemaText);
        expect(avroToTypeScript(schema as RecordType)).toMatchSnapshot();
    });

    test("it should correctly type strings with logicalType", () => {
        const schema: Schema = {
            type: "record",
            name: "datedRecord",
            fields: [
                {
                    name: "eventDate",
                    type: {
                        type: "string",
                        logicalType: "iso-datetime",
                    },
                },
            ],
        };
        expect(avroToTypeScript(schema)).not.toEqual(expect.stringContaining("UNKNOWN"));
    });
});

describe("enumToTypesScript", () => {
    test("it should generate an enum", () => {
        const schemaText = fs.readFileSync(__dirname + "/just-enum.avsc", "utf-8");
        const schema = JSON.parse(schemaText);
        expect(avroToTypeScript(schema as RecordType)).toMatchSnapshot();
    });

    test("it should correctly type strings with logicalType", () => {
        const schema: Schema = {
            type: "enum",
            name: "CompanySize",
            symbols: ["SMALL", "MEDIUM", "LARGE"],
        };

        expect(avroToTypeScript(schema)).not.toEqual(expect.stringContaining("UNKNOWN"));
    });
});
