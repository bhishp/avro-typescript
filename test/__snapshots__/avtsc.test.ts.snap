// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avroToTypeScript it should generate an interface 1`] = `
"export enum numberEnumType { ONE, TWO };

export interface AllocsType {
	allocAccount?: null | undefined | string;
	noNestedPartyIDs?: null | undefined | number;
	allocQty?: null | undefined | number;
}

export interface ExampleType {
	unionEnum?: null | undefined | string | numberEnumType;
	mandatoryString: string;
	adouble?: null | undefined | number;
	astring?: null | undefined | string;
	amap?: null | undefined | string | { [index:string]:numberEnumType };
	recordArray?: null | undefined | AllocsType[];
	processCode?: null | undefined | string;
	named?: null | undefined | AllocsType;
}
"
`;

exports[`enumToTypesScript it should generate an enum 1`] = `
"export enum CompanySize { SMALL, MEDIUM, LARGE };
"
`;
